File: 프로그래머스/2/131127. 할인 행사/할인 행사.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

int solution(char* want[], size_t want_len, int number[], size_t number_len, char* discount[], size_t discount_len) {
    int answer = 0;
    
    int total = 0;
    for (size_t i = 0; i < number_len; i++) {
        total += number[i];
    }

    for (size_t i = 0; i <= discount_len - total; i++) {
        int temp_counts[10] = {0}; 

        for (int j = 0; j < total; j++) {
            char* target = discount[i + j];

            for (size_t k = 0; k < want_len; k++) {
                if (strcmp(target, want[k]) == 0) {
                    temp_counts[k]++;
                    break;
                }
            }
        }

        bool is_match = true;
        for (size_t k = 0; k < want_len; k++) {
            if (temp_counts[k] != number[k]) {
                is_match = false;
                break;
            }
        }

        if (is_match) answer++;
    }

    return answer;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    wantdict = dict(zip(want, number))
    
    total = sum(number)
    answer = 0
    for i in range(len(discount) - total + 1):
        tempdict = {}
        for j in range(total):
            target = discount[i + j]
            tempdict[target] = tempdict.get(target, 0) + 1
        
        is_TF = 1
        for j in want:
            if wantdict[j] != tempdict.get(j, 0):
                is_TF = 0
                break
                
        if is_TF:
            answer += 1
            
    return answer